src/block/Hopper.php

```diff
@@ -99,5 +100,12 @@ public function onScheduledUpdate() : void{
 		//TODO
 	}
 
+	public function getSupportType(int $facing) : SupportType{
+		if($facing === Facing::UP){
+			return SupportType::EDGE();
```

Comment:
It's unclear what the correct behaviour of this should be. Many blocks (e.g. torches, lanterns) can be placed on top of a hopper, but can't be placed on top of other EDGE blocks (such as composters); however, rails can be placed on both.

I think the correct thing for hopper would be to treat the top face as FULL. I don't like it, but it seems to be how the game itself treats it.